src/Query/Transformer/NamedArrayParameter.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Linio\Component\Database\Query\Transformer;
+
+use Linio\Component\Database\Exception\InvalidQueryException;
+use Linio\Component\Database\Query\Builder;
+use Linio\Component\Database\Query\Transformer;
+
+class NamedArrayParameter implements Transformer
+{
+    /**
+     * @var Builder
+     */
+    protected $builder;
+
+    public function __construct(Builder $builder)
+    {
+        $this->builder = $builder;
+    }
+
+    public function execute(string &$query, array &$params = []): void
+    {
+        foreach ($params as $paramKey => $paramValue) {
+            if (is_numeric($paramKey)) {
+                if (is_array($paramValue)) {
+                    throw new InvalidQueryException(sprintf(
+                        'Unnamed parameter "%s" can\'t have array value: %s',
+                        $paramKey,
+                        var_export($paramValue, true)
+                    ));
+                }
+
+                continue;
+            }
+
+            if (is_array($paramValue)) {
+                $placeholders = $this->builder->placeholders($paramKey, $paramValue);
+                $params += $placeholders;
```

Comment:
Here's an example of the dependency: you're doing part of the logic inside the builder, which is not really useful for anyone else, and the other part here. What was the design reasoning behind the split between the too? Where does Builder end and NamedArrayParameter start?

Reply:
you're right, it's a premature optimization to make placeholders() available to any other transformer. thanks for your feedback.